src/libs/CardUtils.ts

```diff
@@ -87,6 +87,34 @@ function getCardDescription(cardID?: number, cards: CardList = allCards) {
     return cardDescriptor ? `${humanReadableBankName} - ${cardDescriptor}` : `${humanReadableBankName}`;
 }
 
+type DomainFeedData = {bank: string; domainName: string; correspondingCardIDs: string[]; fundID?: string};
+
+/**
+ * @param cardList - The list of cards to process. Can be undefined.
+ * @returns a record where keys are domain names and values contain domain feed data.
+ */
+
+function generateDomainFeedData(cardList: CardList | undefined): Record<string, DomainFeedData> {
```

Comment:
does this function return Cards or Feeds?
if feeds then shouldnt it go into the new file FeedUtils?